### ISSUE TITLE
BCXDHPublicKey cannot be cast to class XECPublicKey

### ISSUE THREAD

**Issue by tresf** [qzind/tray#894]
After packaging and deploying QZ Tray 2.2.0+6 to a customer, the following exception occurred:

> ```
> javax.net.ssl.SSLException: class org.bouncycastle.jcajce.provider.asymmetric.edec.BCXDHPublicKey cannot be cast to class java.security.interfaces.XECPublicKey (org.bouncycastle.jcajce.provider.asymmetric.edec.BCXDHPublicKey is in unnamed module of loader 'app'; java.security.interfaces.XECPublicKey is in module java.base of loader 'bootstrap')
> ```

Java version: 11.0.12
Java vendor: Eclipse Foundation (Adoptium/Temurin)

Investigation points to the following:

* https://github.com/bcgit/bc-java/issues/589
* https://github.com/bcgit/bc-java/issues/881

Note, the Windows installer was cross-packaged using `ant nsis -Dtarget.arch=x86_64` an M1 Mac, if that matters.

https://github.com/tresf/tray/releases/download/TEMP/qz-tray-2.2.0+6.exe

This build may have been packaged with `pdfbox-app-2.0.24.jar` containing a different BouncyCastle release.

**Comment by Vzor-** [comment]
I have tested 2 version of pdfbox

pdfbox 2.0.24, containing BC 1.64
pdfbox 3.0.0-alpha2, containing BC 1.69

With both of these the issue persisted
I tested 3 JDKs

11.0.2+7
11.0.12+7
16.0.2+7

the issue does not present on 11.0.2+7
To test with this version, I manually set JLink.downloadJdk `url = "https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.2%2B7/OpenJDK11U-jdk_x64_windows_hotspot_11.0.2_7.zip"` 

An alternitive fix that has worked for all test cases is `System.setProperty("jdk.tls.namedGroups", "secp256r1, secp384r1, ffdhe2048, ffdhe3072");`

Side note, an error occurs on my machine when using pdfbox 3. whenever the app is launched via the exe, a permission error occurs. manually launching the program with runtime/bin/java -jar qz-tray.jar works.
